packages/swap/src/routes/swap.ts

```diff
@@ -160,14 +136,18 @@ router.get(
         include: swapInclude,
         // just get the last one for now
         orderBy: { nativeId: 'desc' },
+        take: 1,
       });
       if (!swap) {
         failedSwap = await prisma.failedSwap.findFirst({
           where: { depositTransactionRef: id },
+          include: { swapDepositChannel: { include: depositChannelInclude } },
         });
       }
     }
 
+    swapDepositChannel ??= swap?.swapDepositChannel ?? failedSwap?.swapDepositChannel;
```

Comment:
would be nice to check this in one of the tests

Reply:
just added it